spec/models/course/condition/achievement_spec.rb

```diff
@@ -34,5 +34,28 @@
         expect(subject.title).to eq(subject.achievement.title)
       end
     end
+
+    describe '#satisfied_by?' do
+      let(:achievement1) { create(:achievement) }
+      let(:achievement2) { create(:achievement) }
+      let(:course_user) do
+        achievements = double
+        allow(achievements).to receive(:exists?).with(achievement1).and_return(true)
+        allow(achievements).to receive(:exists?).with(achievement2).and_return(false)
+        course_user = double
+        allow(course_user).to receive(:achievements).and_return(achievements)
+        course_user
+      end
+
+      it 'returns true if the use has the achievement' do
+        subject.achievement = achievement1
+        expect(subject.satisfied_by?(course_user)).to be_truthy
+      end
+
+      it 'returns false if the context subject does not have the achievement' do
```

Comment:
actually, this should be a context block. `context 'when the user does not have the achievement'`